test/test-unit/Makefile

```diff
@@ -0,0 +1,16 @@
+BIN_DIR=../../build/test/test-unit
+TEST_BINARIES=read_bytes.test
+
+all: $(TEST_BINARIES)
+
+read_bytes.test: read_bytes.c unit.c
+	gcc -o $@ $^
```

Comment:
Just side note: we can add any executable as a unit test, including one that is written on a scripting language (Bourne shell, Python). Only exit code, stdout and stderr are important for test-run.

So maybe it would be easier to verify different things around unit tests using scripts.

Reply:
(`printf` — not `echo` — is for bash / dash portability.)

Reply:
(Sergey pointed me that printf in dash [does not support](https://man7.org/linux/man-pages/man1/dash.1.html) hexadecimal byte values. We can use octal ones instead: `printf '\300\000\n'`.)